lib/src/builders/sentinels.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:nyxx/src/builders/image.dart';
+import 'package:nyxx/src/models/channel/types/forum.dart';
+import 'package:nyxx/src/models/snowflake.dart';
+
+// ASCII encoded "nyxx"
+const sentinelInteger = 0x6E797878;
+
+// ESC-"nyxx"
+const sentinelString = '\u{1B}nyxx';
+
+const sentinelDuration = _SentinelDuration();
+
+class _SentinelDuration implements Duration {
+  const _SentinelDuration();
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+}
+
+const sentinelSnowflake = _SentinelSnowflake();
+
+class _SentinelSnowflake implements Snowflake {
+  const _SentinelSnowflake();
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+}
+
+const sentinelDefaultReaction = _SentinelDefaultReaction();
+
+class _SentinelDefaultReaction implements DefaultReaction {
+  const _SentinelDefaultReaction();
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+}
+
+const sentinelList = _SentinelList();
+
+class _SentinelList implements List<Never> {
+  const _SentinelList();
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+}
+
+const sentinelImageBuilder = _SentinelImageBuilder();
+
+class _SentinelImageBuilder implements ImageBuilder {
+  const _SentinelImageBuilder();
+
+  @override
+  void noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
```

Comment:
why it needs to be implemented?

Reply:
Sentinels exist to be markers for a given type and not carry any implementation.

We could implement every member to throw `UnimplementedError` or we can use a Dart feature that if a class implements `noSuchMethod`, then it does not need to implement any other members.

This leads to shorter code with ultimately the same result, as no method should ever be called on a sentinel instance.